src/VisualStudio/Core/Def/Implementation/ColorSchemes/ColorSchemeApplier.Settings.cs

```diff
@@ -64,20 +67,39 @@ public void ApplyColorScheme(SchemeName schemeName, ImmutableArray<RegistryItem>
                 {
                     using var itemKey = registryRoot.CreateSubKey(item.SectionName);
                     itemKey.SetValue(item.ValueName, item.ValueData);
+                    itemKey.Flush();
                 }
 
-                _workspace.SetOptions(_workspace.Options.WithChangedOption(ColorSchemeOptions.AppliedColorScheme, schemeName));
+                registryRoot.Flush();
+
+                SetAppliedColorScheme(schemeName);
 
                 // Broadcast that system color settings have changed to force the ColorThemeService to reload colors.
                 NativeMethods.PostMessage(NativeMethods.HWND_BROADCAST, NativeMethods.WM_SYSCOLORCHANGE, wparam: IntPtr.Zero, lparam: IntPtr.Zero);
             }
 
+            /// <summary>
+            /// Get the color scheme that is applied to the configuration registry.
+            /// </summary>
             public SchemeName GetAppliedColorScheme()
             {
-                var schemeName = _workspace.Options.GetOption(ColorSchemeOptions.AppliedColorScheme);
-                return schemeName != SchemeName.None
-                    ? schemeName
-                    : ColorSchemeOptions.AppliedColorScheme.DefaultValue;
+                // The applied color scheme is stored in the configuration registry with the color theme information because
+                // when the hive gets rebuilt during upgrades, we need to reapply the color scheme information.
+                using var registryRoot = VSRegistry.RegistryRoot(_serviceProvider, __VsLocalRegistryType.RegType_Configuration, writable: false);
+                using var itemKey = registryRoot.OpenSubKey(ColorSchemeApplierKey);
+                return itemKey is object
+                    ? (SchemeName)itemKey.GetValue("AppliedColorScheme")
```

Comment:
`"AppliedColorScheme"` should likely be a constant since it's used in two places :) Would be really bad if there was a typo making them different.